Cellular: Added API for setting default parameters to a network inter…

mirelachirica opened the pull request:
### Description
Added API for setting default parameters to a network interface.

Reason:
A network interface instantiated directly is initially unconfigured. This API can be used to set the default parameters that would have been set if the interface had been requested using NetworkInterface::get_default_instance().

This is needed when instantiating NonIP cellular interface for example and NetworkInterface::get_default_instance() cannot be used.


### Pull request type
    [x ] Fix
    [ ] Refactor
    [ ] Target update
    [ ] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

Commit:
cellular: eps ciot optimization network support check

-added an API for checking network eps ciot optimization support
-renamed the API for getting the UE parameters
-the API for setting the UE parameters includes now a callback, which
will be called once network support for eps ciot optimization is known

Commit:
cellular: eps ciot optimization network support check astyle fix

Commit:
Cellular: Non-IP socket and PDP context for EPS control plane data delivery

Commit:
Cellular: Move eDRX setting from power to network

Commit:
Cellular: Move PSM setting from power to device

Commit:
Cellular: Fixed CellularStateMachine to stop on all target states.

Commit:
Removed CellularSIM interface.
Moved methods to classes CellularDevice and CellularInformation.
SIM interface was removed to simplify cellular usage and
methods better suite new classes.
Updated greentea and unit tests.

Commit:
Cellular: Move device_ready from power to device

Commit:
Cellular: AT remove_urc_handler changed to set_urc_handler(prefix, 0)

Commit:
Cellular: Move ready_cb from power to device

Commit:
Cellular: Unify set_at_mode and init_module into init()

Commit:
Cellular: Removed set_power_level()

Commit:
Cellular: Moved reset from power to device

Commit:
Cellular: Added shutdown()

Commit:
Cellular: Removed get_extended_signal_quality and changed get_signal_quality

Commit:
Cellular: fix possible crash in state machine

_sim_pin was changed to pointer from array and length was checked with
strlen. If _sim_pin was null it caused crash. Fix by checking _sim_pin against NULL.
Power class could have been called without checking if power is NULL. Fix by checking
that power class is not null.
Fix state machine to return correct states when queried.

Commit:
Cellular: Add State machine unit tests.

Commit:
Cellular: added setting of data carrier support for UART.

Commit:
Cellular: changed support features to CellularProperty array.

Commit:
Merge pull request #8891 from jarvte/remove_sim_interface

Cellular: Remove sim interface

Commit:
Merge pull request #8979 from TeemuKultala/cellular_ciot_eps_on_feature_branch

Cellular ciot eps on feature branch

Commit:
Cellular: Change AT_CellularNetwork to use CellularProperties.

This change enables removing function has_registration from
class AT_CellularNetwork and all targets inheriting
AT_CellularNetwork.

Commit:
Cellular: Remove target files inheriting from AT_CellularNetwork

After AT_CellularNetwork::has_registration was replaced with
CellularProperties and better
AT_CellularNetwork::set_access_technology_impl default
implementation we can delete most of the target specific classes
that inherit AT_CellularNetwork.

Commit:
Merge pull request #8901 from AriParkkila/cell-refactor-power-api

Cell refactor power api

Commit:
Merge pull request #9050 from jarvte/support_feature_to_key_value

Cellular: changed support features to CellularProperty array.

Commit:
Cellular: change stack_type_supported to get_property

Change usage of AT_CellularContext::stack_type_supported to
AT_CellularBase::get_property. This way we can rid of
targets overriding stack_type_supported and delete
unnecessary classes and simplify new targets.

Commit:
Cellular: Removed unnecessary checks after new

After this change we were able to change methods
ATHandler::set_urc_handler and CellularDevice::set_ready_cb to void
and simplify error handling.

Commit:
Merge pull request #9090 from jarvte/remove_new_null_checks

Cellular: Removed unnecessary checks after new

Commit:
Cellular: Added generic cellular modem

Generic cellular module (GENERIC_AT3GPP) can by used as a default
module when porting new cellular module. It's a good starting point
and eases porting of new modules. GENERIC_AT3GPP uses only standard
3GPP AT commands when communicating with the modem.

0xc0170 committed:
Merge pull request #9120 from jarvte/create_generic_modem

Cellular: Add generic cellular modem

0xc0170 committed:
Merge pull request #8971 from jarvte/add_datacarrier_detect

Cellular: added setting of data carrier support for UART.

Commit:
Merge pull request #9121 from mirelachirica/feature-cellular-refactor_nonip

Cellular: Non-IP socket and PDP context for EPS control plane data de…

Commit:
Cellular: Unit tests for Non-IP socket

0xc0170 committed:
Merge pull request #9226 from mirelachirica/feature-cellular-refactor_nonip_ut

Cellular: Unit tests for Non-IP socket

ciarmcom commented:
@mirelachirica, thank you for your changes.
@ARMmbed/mbed-os-ipcore @ARMmbed/mbed-os-wan @ARMmbed/mbed-os-maintainers please review.

kjbracey reviewed (commented):
This is also adding `CellularContext::get_default_non_ip_instance()`, not mentioned in the description.

Core changes seem fine, but @SeppoTakalo should comment.

SeppoTakalo commented:
This looks OK for me.

As a side effect, now anyone can overwrite the ´set_default_parameter()` function, but maybe that is OK.

NirSonnenschein commented:
The Travis failures are caused by an issue fixed on master a few hours ago, and this change should be propagated here via rebase. We can start CI only after travis is fixed, could you please rebase this PR
Sorry for the trouble

0xc0170 commented:
Looking at this, unfortunate, but either to add the fix for travis or rebase the destination branch? Might be easier just that travis fix here

Commit:
Fix travis by cherry pick commit 84469e1

Corrected simplified astyle job behavior when base branch is not master

Consolidated efforts from PR #9336.

Previous job's env vars would not be populated correctly if the base branch of a PR was not master.
Corected by pulling remaining respository information to perform comparison between read-only instance of PR and base branch.

cmonr commented:
Note: https://github.com/ARMmbed/mbed-os/pull/7976#issuecomment-453383516

mirelachirica commented:
Can I cherry pick travis fix here or does it need a separate PR?

mirelachirica commented:
> The Travis failures are caused by an issue fixed on master a few hours ago

@NirSonnenschein could you tell the commit you are refering to?

thanks

mirelachirica commented:
> Note: [#7976 (comment)](https://github.com/ARMmbed/mbed-os/pull/7976#issuecomment-453383516)

this refers to fix in:
https://github.com/ARMmbed/mbed-os/pull/9346
but this PR is not merged. 
@NirSonnenschein reffers to a fix that is in master already. Which one is it?

0xc0170 commented:
Lets wait for 9346, should be in within today

0xc0170 commented:
It's on master now

mirelachirica commented:
> It's on master now

I'm not sure about the cherry picking commit. It ran into conflict because travis file was many commits behind. I took the content as it is now in master. Is that ok?

0xc0170 commented:
yes, should be

cmonr commented:
Trying a thing. Standby.

cmonr commented:
This is getting _really_ annoying...
We've been seeing more and more commits **not** get triggered in the past couple of days. Not sure if the problem is with GitHub or Travis CI...

0xc0170 commented:
This will go into CI once 5.11.2 RC test completes (soon in the current queue)

mirelachirica commented on UNITTESTS/stubs/NetworkInterfaceDefaults_stub.cpp:
```diff
@@ -0,0 +1,29 @@
+#include "netsocket/NetworkInterface.h"
+#include "CellularBase.h"
+
+void NetworkInterface::set_default_parameters()
```
These two need to be removed from here and use NetworkInterfaceDefaults_stub.cpp

0xc0170 commented:
CI started

Commit:
Cellular: Added API for setting default parameters to a network interface

A network interface instantiated directly is initially unconfigured.
This API can be used to set the default parameters that would
have been set if the interface had been requested using
NetworkInterface::get_default_instance()

mirelachirica commented:
Fixed the CI UT linking error

0xc0170 commented:
CI started

0xc0170 commented:
Latest rebase caused few conflicts here.

mirelachirica commented:
This PR was added to feature branch during rebase by Ari:
"The remaining PRs #9273 and #9287 going to the feature-branch can be closed, they were already approved so I added them in the feature-branch as well."

Based on this I will close this PR.
